.github/workflows/verify.yaml

```
@@ -0,0 +1,76 @@
+name: verify
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+
+      - name: Install ShellCheck and shfmt
+        run: |
+          sudo apt update
+          sudo apt install -y shellcheck
+          go install mvdan.cc/sh/v3/cmd/shfmt@latest
+
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
+          only-new-issues: true
+          args: --timeout=5m
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: codespell
+        uses: codespell-project/actions-codespell@v2
+        with:
+          skip: vendor,*.svg
+      
+      - name: ShellCheck and shfmt for wait_for_deployment.sh
+        run: |
+          shellcheck ./test/wait_for_deployment.sh
+          shfmt -d ./test/wait_for_deployment.sh
```

```suggestion
      - name: ShellCheck and shfmt
        run:
          shellcheck ./test/*.sh
          shfmt -d ./test/*.sh
```
We should be running it on a general pattern since we may add more files going forward.